src/identity/masterpubkey.ts

```diff
@@ -276,70 +270,60 @@ export class MasterPublicKey extends Identity implements IdentityInterface {
 
   private async restoreAddresses(): Promise<AddressInterfaceExtended[]> {
     const NOT_USED_ADDRESSES_LIMIT = 20;
-    let counter = 0;
-    let index = 0;
-
-    const restoredAddresses: AddressInterfaceExtended[] = [];
+    let restoredAddresses: AddressInterfaceExtended[] = [];
+
+    for (let i = 0; i < 2; i++) {
+      const change = i === 1;
+      let counter = 0;
+      let index = 0;
+
+      const usedAddresses: AddressInterfaceExtended[] = [];
+      const incrementOrResetCounter = (
+        addresses: AddressInterfaceExtended[]
+      ) => (hasBeenUsed: boolean, index: number) => {
+        if (hasBeenUsed === true) {
+          counter = 0;
+          usedAddresses.push(addresses[index]);
+        } else {
```

Comment:
else branching cound be dropped. We always increment the count, if we enter in the `if hasBeenUsed` then we overwrite the counter to 0 anyway